src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/GaugeWave.cpp

```diff
@@ -0,0 +1,151 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/GaugeWave.hpp"
+
+#include <cmath>  // IWYU pragma: keep
+#include <cstdlib>
+#include <numeric>
+#include <ostream>
+#include <pup.h>
+#include <utility>
+
+#include "DataStructures/DataBox/Prefixes.hpp"  // IWYU pragma: keep
+#include "DataStructures/DataVector.hpp"        // IWYU pragma: keep
+#include "DataStructures/TempBuffer.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Parallel/PupStlCpp11.hpp"
+#include "PointwiseFunctions/GeneralRelativity/ComputeSpacetimeQuantities.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/StdArrayHelpers.hpp"
+#include "Utilities/StdHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+
+// IWYU pragma: no_include <complex>
+
+// IWYU pragma: no_forward_declare Tensor
+
+/// \cond
+namespace {
+// compute gauge wave H
+template <typename DataType>
+Scalar<DataType> gauge_wave_h(const tnsr::I<DataType, 3>& x, const double t,
+                              const double amplitude, const double wavelength) {
+  auto h = make_with_value<Scalar<DataType>>(x, 0.0);
+  get(h) =
+      -1.0 * amplitude * sin((get<0>(x) - t) * (2.0 * M_PI / wavelength)) + 1.0;
```

Comment:
You don't need to `make_with_value` and then overwrite it. Just `return {-1. * amplitude * ...};` so the Scalar will be constructed from the correct DataVector directly. (same in deriv_h, but there you probably want to return the DataVector directly since it's only a component of a covector field)

Reply:
I thought we have a general policy of passing/returning Scalar<DataVector> over plain DataVectors.

Reply:
I now realize what's going on here: you can construct a DataVector with {}, but not a Scalar<DataVector>, because the Tensor constructor is marked explicit. I suppose it isn't a big deal for my anonymous namespace functions to pass DataVector instead of Scalar<DataVector>, so I'll make that change to avoid the make_with_value<> call as you suggested, @nilsleiffischer.